### DISCUSSION TITLE
Bash - Nemo selections

### DISCUSSION THREAD

**Discussion by Jopp-gh** [linuxmint/discussions#584] [Ideas]
Nemo and gio have some nice cli commands, still one of the key features is missing:

"To return selected files or/and folders as new-line delimited list items"

I've discussed this topic already here:
https://forums.linuxmint.com/viewtopic.php?t=423388

The more I use Mint, the more I'm happy about it's scripting capabilities.

Still, I'm puzzled why Nemo needs a complicated VAR called NEMO_SCRIPT_SELECTED_FILE_PATHS which doesn't work outside the contextual menu of "action scripts". Well,  a simple "$1" is enough, what's the point to use NEMO_SCRIPT_SELECTED_FILE_PATHS ?

Is there some command I'm missing which is able to return selected files/ folders without the context menu?  

If no, I would suggest the developers of Mint to expand Nemo's scripting vocabulary and add Nemo file / folder selections too ( reference is the active split view of window 1 )

Nemo selections as command would be very much appreciated.

**Comment by mtwebster** [comment]
Hi, Nemo Actions are sort of a replacement for scripts (we only still support scripts for backward compatibility.

https://github.com/linuxmint/nemo/blob/master/files/usr/share/nemo/action-info.md

You should be able to accomplish what you want with them.

**Reply by Jopp-gh** [reply]
Thanks for your kind answer, I understand. 
Yeah, Nemo actions are really great!

My current problem is, I want to zip/unzip files / folders with a simple shortcut combo. I know there's an entry for compressing files and it should be possible to add a shortcut to this default action, but I've to research this first. Anyway, I don't know if there's an equivalent to unzip files.(without GUI interaction)

Anyway, if we could get Nemo selections, bash scripts outside of context action scripts would be so more useful and awesome.

**Comment by Jopp-gh** [comment]
I found something interesting. https://forums.linuxmint.com/viewtopic.php?t=344193

The idea to open zip files automatically doesn't even need a shortcut, if Nemo is configured to open folders and files on single click.
Still: selecting folders/files and then, execute a shortcut combo to zip selected items in Nemo active window, would require a "Nemo selection". The only possible way to zip regular files is through the context menu.

I'd like to close this idea as resolved, but my point was to request a new option for Nemo's CLI interface. I would appreciate a "get selected files or folders" feature in Nemo.

**Comment by mtwebster** [comment]
In the long run I think allowing nemo actions to have shortcuts would be more useful to more people. What you're after would still be possible.
